addons/stock_account/models/stock_move.py

```diff
@@ -377,6 +389,55 @@ def _prepare_account_move_line(self, qty, cost, credit_account_id, debit_account
 
         return res
 
+    def _prepare_analytic_line(self):
+        self.ensure_one()
+        if not self._get_analytic_account():
+            return False
+
+        if self.state in ['cancel', 'draft']:
+            return False
+
+        if self.state != 'done':
+            unit_amount = self.product_uom._compute_quantity(
+                self.quantity_done, self.product_id.uom_id)
+            # Falsy in FIFO but since it's an estimation we don't require exact correct cost. Otherwise
+            # we would have to recompute all the analytic estimation at each out.
+            amount = -unit_amount * self.product_id.standard_price
+        elif self.product_id.valuation == 'real_time':
+            accounts_data = self.product_id.product_tmpl_id.get_product_accounts()
+            account_valuation = accounts_data.get('stock_valuation', False)
+            analalytic_line_vals = self.stock_valuation_layer_ids.account_move_id.line_ids.filtered(
+                lambda l: l.account_id == account_valuation)._prepare_analytic_line()
+            amount = sum(vals['amount'] for vals in analalytic_line_vals)
+            unit_amount = sum(vals['unit_amount']
+                              for vals in analalytic_line_vals)
+        elif sum(self.stock_valuation_layer_ids.mapped('quantity')):
+            amount = sum(self.stock_valuation_layer_ids.mapped('value'))
+            unit_amount = amount / \
+                sum(self.stock_valuation_layer_ids.mapped('quantity'))
```

Comment:
@amoyaux unit_amount is just the quantity, no?

Reply:
Yep gonna modify it